src/core/sock/sock-redirect.cpp

```diff
@@ -306,115 +309,154 @@ bool handle_close(int fd, bool cleanup, bool passthrough)
     return to_close_now;
 }
 
-#if defined(DEFINED_NGINX)
+#if defined(DEFINED_NGINX) || defined(DEFINED_ENVOY)
 
-int init_child_process_for_nginx()
+static int init_worker(int worker_id, int listen_fd)
```

Comment:
Please consider splitting this method to smaller units: handle_envoy_parent/handle_tcp_socket/handle_udp_socket
Such long methods are very hard to read and understand.